### ISSUE TITLE
AsyncJavaRedis#remove - Command DEL for key 'XXXX' failed.

### ISSUE THREAD

**Issue by master-elodin** [KarelCemus/play-redis#230]
### Reporting a bug?
1. Version of play-redis
2.5.1
1. Version of Scala
2.12.10
1. Version of Play
2.7.4
1. Describe the environment if applies (usually in cloud etc.)
AWS
1. Describe the expected behavior
Removing a key should remove the key and the classTagKey when there are multiple Redis nodes
1. Describe the actual behavior, follow the same scenario
Removing a key usually (but not always) throws an exception `Command DEL for key 'XXXX' failed`
1. Provide instructions to reproduce the issue
* Create multi-node Redis cluster
* Java server using `AsyncJavaRedis` (via `AsyncCacheApi`) set a value for key `mykey` - `asyncCacheApi.set('mykey', myObject);`
* Remove key - `asyncCacheApi.remove('mykey');`
* Exception is thrown and key is not deleted (although async operation returns a success)
  * `play.api.cache.redis - Command DEL for key 'mykey classTag::mykey' failed.`
    * `redis.actors.ReplyErrorException: CROSSSLOT Keys in request don't hash to the same slot`

---

I think this is happening because the key and the classTagKey are both being removed in the same operation but they're not guaranteed to be on the same Redis node.
https://stackoverflow.com/a/38042895

```scala
def remove(key: String): CompletionStage[Done] = {
  internal.remove(key, classTagKey(key)).asJava
}
```

I think removing the keys in separate operations (which is already being done in `AsyncJavaRedis.removeMatching`) will solve the issue.

---

#### Proposed fix
```scala
def remove(key: String): CompletionStage[Done] = {
  async { implicit context =>
    Future.from(
      internal.remove(key),
      internal.remove(classTagKey(key))
    ).asDone
  }
}
```

**Comment by KarelCemus** [comment]
Hey @master-elodin, thanks for the report and detailed issue analysis. This sounds very reasonable, let's do it. I'll take a look at this hopefully later today and release a new version.

**Comment by master-elodin** [comment]
@KarelCemus I've got a PR open with the fix... it's my first open source PR though, so let me know if I did anything wrong!

**Comment by KarelCemus** [comment]
Hey there, sorry for the delay. You did a great job, thanks for your contribution. Your PR is perfect, I just added a note into the changelog file and merging your changes. I'll release it as `2.6.1`

**Comment by KarelCemus** [comment]
Released `2.6.1`
